rust/src/tor.rs

```diff
@@ -81,6 +82,34 @@ pub struct LwdConn {
 }
 
 impl LwdConn {
+    /// Returns information about this lightwalletd instance and the blockchain.
+    pub(crate) fn get_lightd_info(&mut self) -> anyhow::Result<service::LightdInfo> {
+        Ok(self
+            .runtime
+            .clone()
+            .block_on(async { self.conn.get_lightd_info(service::Empty {}).await })?
```

Comment:
Rust question: it's surprising to me that the `async` and `await` are required here. My mental model of async functions is that they're just syntactic sugar over functions that return their results in a `Future`; if that were the case though, with `.block_on` taking an argument of type `Future<A>` and `.get_lightd_info` returning a Future, why the need to wrap the call?

Reply:
You're correct that this could be simplified. I'm just in the habit of the `async {}` wrapping because usually I'm invoking more than one async method.